File: Calculator/app/src/main/java/com/example/calculator/MainActivity.java
package com.example.calculator;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private Button btn;
    private EditText et1,et2;
    private TextView result;
    private RadioGroup rg;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //아이디 값 찾아주기
        btn=(Button)findViewById(R.id.btn);
        et1=(EditText)findViewById(R.id.et1);
        et2=(EditText)findViewById(R.id.et2);
        rg=(RadioGroup)findViewById(R.id.rg);
        result=(TextView)findViewById(R.id.result);

        //버튼 리스너
        btn.setOnClickListener(this);
    }


    @Override
    public void onClick(View view) {


        if (et1.getText().toString().equals("") || et2.getText().toString().equals("")) {
            Toast.makeText(getApplicationContext(), "숫자를 입력해주세요", Toast.LENGTH_SHORT).show();
        }else {
            int num1=Integer.parseInt(et1.getText().toString());
            int num2=Integer.parseInt(et2.getText().toString());

            int id = rg.getCheckedRadioButtonId();
            if(id == R.id.rb1) {
                Intent intent = new Intent(MainActivity.this,AddActivity.class);
                intent.putExtra("num1",num1);
                intent.putExtra("num2",num2);
                startActivityForResult(intent,0);
            }else if(id == R.id.rb2) {
                Intent intent = new Intent(MainActivity.this,SubActivity.class);
                intent.putExtra("num1",num1);
                intent.putExtra("num2",num2);
                startActivityForResult(intent,0);
            }else if(id == R.id.rb3) {
                Intent intent= new Intent(MainActivity.this,MulActivity.class);
                intent.putExtra("num1",num1);
                intent.putExtra("num2",num2);
                startActivityForResult(intent,0);
            }else{
                Toast.makeText(MainActivity.this,"버튼을 선택해주세요",Toast.LENGTH_SHORT).show();
            }

        }
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (resultCode == RESULT_OK) {
            int resultValue = data.getIntExtra("resultVal", 0);
            result.setText(String.valueOf(resultValue));
        }
    }
}

File: Kotlin/CaculatorBMI/app/src/main/java/com/example/caculatorbmi/MainActivity.kt
package com.example.caculatorbmi

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val heightEditText = findViewById<EditText>(R.id.et_cm)
        val weightEditText = findViewById<EditText>(R.id.et_kg)
        val submitButton = findViewById<Button>(R.id.btn_submit)

        submitButton.setOnClickListener{
            if(heightEditText.text.isEmpty() || weightEditText.text.isEmpty()){
                Toast.makeText(this, "값을 입력해주세요.",Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }
            val height : Int = heightEditText.text.toString().toInt()
            val weight : Int = weightEditText.text.toString().toInt()

            val intent = Intent(this, ResultActivity::class.java)
            intent.putExtra("height", height)
            intent.putExtra("weight", weight) // intent는 onCreate()에 넣어서 전달한다.
            startActivity(intent)
        }

    }
}